doc/adr/0013-synchronous_peer_discovery.md

```diff
@@ -0,0 +1,73 @@
+However, peer discovery can be a long running process. Therefore, it is important that there continue
+to be non-terminating peer discovery process that runs parrallel to other node processes.
+
+### The process
+
+The current bootstrap process will remain largely unchanged. A new node (_Node A_) will be given an
+existing node (_Node B_) to bootstrap off of. After starting, the two nodes will join each other.
+_Node B_ will be the first peer known by _Node A_. _Node A_ then queries _Node B_ for its list of
+known peers. These newly discovered peers are added to _Node A_'s list of known peers. The peer
+discovery process repeats for each peer added to the queue of discovered peers. This is what the
+current peer discovery process does.
+
+## Proposal
+
+A possible solution to our problem is to create a new separate, peer discovery process that will run
+after joining and before observation. It will not run as an asychronous process and will terminate
+upon hitting a terminating condition (e.g., a time limit, a certain number of peers discovered, etc.).
+It will also only run once, as part of the bootstrapping process.
+
+However, instead of creating a separate peer discovery process, we will simply delay the start of
+observation until a certain number of peers have been discovered.
+
+### Demonstration Implementation
+
+David created a demonstration implementation using `Deferred`. A `Deferred` will be created as part
+of `PeerDiscovery` and observation will have to wait until the `Deferred` is complete to run. As
+part of the process of discovering from a peer, the size of the cache of discovered peers will be
+checked. If it exceeds a given threshhold, the `Deferred` is completed and the observation process
+is allowed to run.
+
+This change only affects the execution of code that choose to wait on that `Deferred`. Other code will
+be unaffected. The `discoverFrom` process will remain mostly unchanged. There will be some
+optimizations to the `discoverFrom` process and the `Deferred` will be updated as a part the
+`discoverFrom` process, but that's it.
+
+A demonstration implementation of using `Deferred` is [here](https://github.com/Constellation-Labs/tessellation/blob/peerDiscovery/modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/PeerDiscovery.scala#L26-L67).
+A demonstration implementation of delaying the observation process is [here](https://github.com/Constellation-Labs/tessellation/blob/peerDiscovery/modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/programs/Download.scala#L30-L34).
+
+## Consequences
+
+Adding a synchronous peer discovery process to the initial node joining process will lengthen that
+node joining process by the _timeout_ amount. For that increased time, we gain some security by
+being able to determine the majority fork with higher confidence instead of completely trusting the
+peer provided at the beginning of the bootstrap process.
+
+## Decision
```

Comment:
To match the template structure, decision should be before consequences

Reply:
Moved.